tests/services/test_criteria_matcher.py

```
@@ -0,0 +1,326 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from app.models import Criteria, CriterionType, Operator
+from app.services.criteria_matcher import CriteriaMatcher
+
+
+@pytest.fixture
+def mock_stat():
+    """Fixture to create a mock stat_result object."""
+
+    def _mock_stat(
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `_mock_stat`

(ANN202)